nodejs/examples/vpc/index.ts

```diff
@@ -55,12 +56,23 @@ const vpcWithLocations = new awsx.ec2.Vpc("custom6", {
     ],
 }, providerOpts);
 
-const vpcWithSpecificZones = new awsx.ec2.Vpc("custom7", {
-    requestedAvailabilityZones: ["us-east-1a", "us-east-1b", "us-east-1c"],
-    subnets: [
-        { type: "public" },
-        { type: "private" },
-        { type: "isolated", name: "db" },
-        { type: "isolated", name: "redis" },
-    ],
-}, { provider: new aws.Provider("prov2", { region: "us-east-1" }) });
+
+export = async () => {
+    const allZones =   await aws.getAvailabilityZones(/*args:*/ undefined, { provider: providerOpts.provider, async: true })
+
+    const selectedZones = new random.RandomShuffle("custom7_az", {
+        inputs: allZones.names,
+        resultCount: allZones.names.length / 2,
```

Comment:
@leezen The documentation doesn't mention what happens if you don't get an integer here, I could floor/ceil it if necessary.